[ESIMD][DOC][NFC] Deprecate DPAS type s1 and u1; clarify DPAS HW reqs (#12412)
intel/llvm: sycl/doc/extensions/supported/sycl_ext_intel_esimd/sycl_ext_intel_esimd.md

consequence of the unit subgroup size is guaranteed independent forward
progress between work-items on many Intel GPU architecture generations.

> Explicit SIMD APIs can be used only in the code to be executed on Intel graphics
> architecture devices. Attempt to run such code on other devices will result in
> error.

Reviewer: ```suggestion
Explicit SIMD APIs can be used only code that will be executed on Intel graphics
architecture devices. Attempting to run such code on other devices will result in an
error.
```

more examples can be found in the
[ESIMD test suite](https://github.com/intel/llvm/tree/sycl/sycl/test-e2e/ESIMD) on github.

## Device queries and conditional dispatching of the code

> ESIMD API provides the access to low level GPU hardware API. At the ESIMD program

Reviewer: ```suggestion
ESIMD API provides access to low level GPU hardware API. At ESIMD program
```

compilation time it is not known what target device is going to be used to run the program.

> ESIMD programming model relies on user to manage the corresponding compile- and/or

Reviewer: ```suggestion
The ESIMD programming model relies on the user to manage the corresponding compile- and/or
```

> runtime-checks that prevents ESIMD API from running on GPU that does not support such API.

Reviewer: ```suggestion
runtime-checks to prevent the ESIMD API from running on a GPU that does not support the API.
```

> One of the most trivial way to manage such checks is to have them on HOST. This variant

Reviewer: ```suggestion
One of the most trivial ways to manage such checks is to have them on the HOST. This variant
```

includes a) calling device detect query to understand what device is being used
b) depending on the device run one or another version of the kernel.
For example, [this test](https://github.com/intel/llvm/blob/sycl/sycl/test-e2e/ESIMD/dpas/dpas_int.cpp#L8) is designed to be run on DG2 and PVC even though those two devices have

> different `execution size`. This is done via usage of [device query on HOST](https://github.com/intel/llvm/blob/sycl/sycl/test-e2e/ESIMD/dpas/dpas_common.hpp#L430) and subsequent

Reviewer: ```suggestion
different `execution sizes`. This is done via usage of [device query on HOST](https://github.com/intel/llvm/blob/sycl/sycl/test-e2e/ESIMD/dpas/dpas_common.hpp#L430) and subsequent
```

> different `execution size`. This is done via usage of [device query on HOST](https://github.com/intel/llvm/blob/sycl/sycl/test-e2e/ESIMD/dpas/dpas_common.hpp#L430) and subsequent

Reviewer: ```suggestion
different `execution size`. This is done via usage of [device query on the HOST](https://github.com/intel/llvm/blob/sycl/sycl/test-e2e/ESIMD/dpas/dpas_common.hpp#L430) and a subsequent
```
